From 99ffa6b4a4f311c88b6b38b31e579b2102453f3a Mon Sep 17 00:00:00 2001
From: Stefan Westerfeld <stefan@space.twc.de>
Date: Sat, 28 Jan 2023 20:49:45 +0100
Subject: [PATCH] Make random number generator code compile with clang++-15 +
 libc++-15.

A uniform random bit generator G needs to have a typedef for G::result_type.
See issue #29.

Signed-off-by: Stefan Westerfeld <stefan@space.twc.de>
--- src/random.hh
+++ src/random.hh
@@ -49,7 +49,9 @@ public:
   Random (uint64_t seed, Stream stream);
   ~Random();
 
-  uint64_t
+  typedef uint64_t result_type;
+
+  result_type
   operator()()
   {
     if (buffer_pos == buffer.size())
@@ -57,12 +59,12 @@ public:
 
     return buffer[buffer_pos++];
   }
-  static constexpr uint64_t
+  static constexpr result_type
   min()
   {
     return 0;
   }
-  static constexpr uint64_t
+  static constexpr result_type
   max()
   {
     return UINT64_MAX;
